sdk/storage/azure_storage_blob/src/clients/blob_client.rs

```diff
@@ -195,7 +195,7 @@ impl BlobClient {
     ///
     /// # Arguments
     ///
-    /// * `metadata` - The metadata headers.
+    /// * `metadata` - A `HashMap` containing the metadata key-value pairs to set for the blob.
```

Comment:
Nit (can fix it in the next PR; other places, too): use links when first mentioning types other than yourself (even within the same crate):


```suggestion
    /// * `metadata` - A [`HashMap`] containing the metadata key-value pairs to set for the blob.
```